test/unit-test/FreeRTOS_RA/FreeRTOS_RA_utest.c

```diff
@@ -0,0 +1,1082 @@
+
+    TEST_ASSERT_EQUAL( 0, xEndPoint.xRAData.bits.bIPAddressInUse );
+}
+
+/**
+ * @brief This function verify received Neighbour Advertisement message
+ *        to see that the chosen IP-address is not in use.
+ */
+void test_vReceiveNA_bIPAddressNotInUse1( void )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer;
+    NetworkEndPoint_t xEndPoint;
+    ICMPPacket_IPv6_t xICMPPacket;
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxNetworkBuffer->pucEthernetBuffer = ( uint8_t * ) ( uint8_t * ) &xICMPPacket;
+    xEndPoint.bits.bWantRA = pdFALSE_UNSIGNED;
+
+    FreeRTOS_FirstEndPoint_ExpectAnyArgsAndReturn( &xEndPoint );
+
+    FreeRTOS_NextEndPoint_ExpectAnyArgsAndReturn( NULL );
+
+    vReceiveNA( pxNetworkBuffer );
+
+    TEST_ASSERT_EQUAL( 0, xEndPoint.xRAData.bits.bIPAddressInUse );
+}
+
+/**
+ * @brief This function verify received Neighbour Advertisement message
+ *        to see that the chosen IP-address is not in use.
+ */
+void test_vReceiveNA_bIPAddressNotInUse2( void )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer;
+    NetworkEndPoint_t xEndPoint;
+    ICMPPacket_IPv6_t xICMPPacket;
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxNetworkBuffer->pucEthernetBuffer = ( uint8_t * ) ( uint8_t * ) &xICMPPacket;
+    xEndPoint.bits.bWantRA = pdTRUE_UNSIGNED;
+
+    FreeRTOS_FirstEndPoint_ExpectAnyArgsAndReturn( &xEndPoint );
+    FreeRTOS_NextEndPoint_ExpectAnyArgsAndReturn( NULL );
+    vDHCP_RATimerReload_Ignore();
+
+    vReceiveNA( pxNetworkBuffer );
+
+    TEST_ASSERT_EQUAL( 0, xEndPoint.xRAData.bits.bIPAddressInUse );
+}
+
+/**
+ * @brief This function verify received Neighbour Advertisement message
+ *        to see that the chosen IP-address is not in use.
+ */
+void test_vReceiveNA_bIPAddressNotInUse3( void )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer;
+    NetworkEndPoint_t xEndPoint;
+    ICMPPacket_IPv6_t xICMPPacket;
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxNetworkBuffer->pucEthernetBuffer = ( uint8_t * ) ( uint8_t * ) &xICMPPacket;
+    xEndPoint.xRAData.eRAState = eRAStateWait;
+    xEndPoint.bits.bWantRA = pdTRUE_UNSIGNED;
+    xEndPoint.xRAData.eRAState = eRAStateIPWait;
+
+    FreeRTOS_FirstEndPoint_ExpectAnyArgsAndReturn( &xEndPoint );
+    FreeRTOS_NextEndPoint_ExpectAnyArgsAndReturn( NULL );
+
+    vReceiveNA( pxNetworkBuffer );
+
+    TEST_ASSERT_EQUAL( 0, xEndPoint.xRAData.bits.bIPAddressInUse );
+}
+
+/**
+ * @brief This function verify received Neighbour Advertisement message
+ *        to see that the chosen IP-address is not in use.
+ */
+void test_vReceiveNA_bIPAddressNotInUse4( void )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer, xNetworkBuffer;
+    NetworkEndPoint_t xEndPoint;
+    ICMPPacket_IPv6_t xICMPPacket;
+    IPv6_Address_t xIPAddress;
+
+    pxNetworkBuffer = &xNetworkBuffer;
+    pxNetworkBuffer->pucEthernetBuffer = ( uint8_t * ) ( uint8_t * ) &xICMPPacket;
+    xEndPoint.xRAData.eRAState = eRAStateWait;
```

Comment:
This UT fails in the assert check 
bWantRA and eRAState can be defined as the above test case , and the assert check should be with 1, as its assigned TRUE [here ](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/dev/IPv6_integration/source/FreeRTOS_RA.c#L240)

Reply:
Which ASSERT check?